android/src/main/kotlin/com/drenther/upi_pay/UpiPayPlugin.kt

```diff
@@ -147,29 +151,46 @@ class UpiPayPlugin internal constructor(registrar: Registrar, channel: MethodCha
     }
   }
 
-  override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?): Boolean {
-    if (requestCodeNumber == requestCode && result != null) {
-      if (data != null) {
-        try {
-          val response = data.getStringExtra("response")!!
-          this.success(response)
-        } catch (ex: Exception) {
-          this.success("invalid_response")
-        }
-      } else {
-        this.success("user_cancelled")
-      }
-    }
-    return true
+
+  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "upi_pay")
+    channel.setMethodCallHandler(this)
+  }
+//
+//  override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?): Boolean {
+//    if (requestCodeNumber == requestCode && result != null) {
+//      if (data != null) {
+//        try {
+//          val response = data.getStringExtra("response")!!
+//          this.success(response)
+//        } catch (ex: Exception) {
+//          this.success("invalid_response")
+//        }
+//      } else {
+//        this.success("user_cancelled")
+//      }
+//    }
+//    return true
+//  }
```

Comment:
Go ahead and remove redundant code instead of commenting them out

Reply:
Yeah it was just for reference purposes 😂 I'll change this too my knowledge of kotlin is not upto the point